File: nekoshlink/shlink-security/src/main/java/org/nekosoft/shlink/sec/delegation/annotation/RunAs.java
package org.nekosoft.shlink.sec.delegation.annotation;

import java.lang.annotation.*;

@Target({ElementType.METHOD, ElementType.CONSTRUCTOR})
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface RunAs {
    String[] roles();
    boolean allowAnonymous() default false;
}

File: nekoshlink/shlink-security/src/main/kotlin/org/nekosoft/shlink/sec/delegation/annotation/RunAs.kt
package org.nekosoft.shlink.sec.delegation.annotation

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.CONSTRUCTOR)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class RunAs(
    vararg val roles: String,
    val allowAnonymous: Boolean = false,
)
